doc/source/data/performance-tips.rst

```diff
@@ -81,14 +116,23 @@ This can be used in conjunction with column pruning when appropriate to get the
 Tuning Read Parallelism
 ~~~~~~~~~~~~~~~~~~~~~~~
 
+By default, Ray Data automatically selects the read ``parallelism`` according to the following procedure:
+
+1. The number of available CPUs is estimated. If in a placement group, the number of CPUs in the cluster is scaled by the size of the placement group compared to the cluster size. If not in a placement group, this is the number of CPUs in the cluster.
+2. The parallelism is set to the estimated number of CPUs multiplied by 2. If the parallelism is less than 8, it is set to 8.
+3. The in-memory data size is estimated. If the parallelism would create in-memory blocks that are larger on average than the target block size (512MiB), the parallelism is increased until the blocks are < 512MiB in size.
+4. The parallelism is truncated to ``min(num_files, parallelism)``.
```

Comment:
Maybe out-of-scope for this PR, but don't think this is totally accurate:
* "If the parallelism is less than 8, it is set to 8." -- don't think this actually happens?
* No mention of the default 200 parallelism
* No mention of `max_reasonable_parallelism`
* "The parallelism is truncated to ``min(num_files, parallelism)``" -- this doesn't make sense for APIs like `read_sql`
 
```
    min_safe_parallelism = 1
    max_reasonable_parallelism = sys.maxsize
    if reader:
        mem_size = reader.estimate_inmemory_data_size()
        if mem_size is not None and not np.isnan(mem_size):
            min_safe_parallelism = max(1, int(mem_size / ctx.target_max_block_size))
            max_reasonable_parallelism = max(
                1, int(mem_size / ctx.target_min_block_size)
            )
    if parallelism < 0:
        avail_cpus = avail_cpus or _estimate_avail_cpus(cur_pg)
        parallelism = max(
            min(ctx.min_parallelism, max_reasonable_parallelism),
            min_safe_parallelism,
            avail_cpus * 2,
        )
        ...
    return parallelism, min_safe_parallelism
```

Instead of describing exactly how this works, maybe it'd be easier to just describe it at a high level?
> Ray Data automatically selects the read ``parallelism`` so that blocks fit in memory and operations are performant."

Reply:
I'm pretty sure this is actually accurate. The algorithm has changed since last...

Definitely out of scope for this PR though, I just moved this block verbatim.